### PR TITLE
Prepare 0.22.9 release

### PR DESCRIPTION
**Pull request by hsyl20**

### CODE REVIEW

**Comment by hsyl20** [comment]
Somehow `cabal-doctest` doesn't find the new version of `doctest`. Isn't it because we pass `--ignore-project`?

**Comment by sol** [comment]
> Somehow `cabal-doctest` doesn't find the new version of `doctest`. Isn't it because we pass `--ignore-project`?

Yes, that's expected.  The `cabal-doctest` workflow will fail whenever you do a release. Build failures for `cabal-coctest` are informational for that reason, they don't block a merge. I don't have a better solution for this yet.

**Comment by hsyl20** [comment]
This MR now includes the fix for the testsuite to pass. See https://github.com/sol/doctest/pull/448#issuecomment-2259952049

**Comment by sol** [comment]
> With the right command line to run tests (I was using cabal test) I only get 1 failure with 9.10:

It's still not clear to me why this failure does not show up on CI.  Is this maybe depending on the value of TERM or something? On the one hand, this would make sense, on the other hand you shouldn't look at TERM at all if the output is not going to a terminal.

**Comment by sol** [comment]
> on the other hand you shouldn't look at TERM at all if the output is not going to a terminal.

The reason why `ghci` prints those links in the first place is because of

https://github.com/sol/doctest/blob/f930eaae60ecdd13a1e933248dbd1c886fbdc399/src/Language/Haskell/GhciWrapper.hs#L64

Redirection of `stderr` is only done after `ghci` has started. At that point GHC already decided that output is going to a terminal, hence we see the links. 

A proper solution would be to do something along the lines of: https://github.com/hspec/sensei/blob/5c11026fa48e13ea1c351ab882765eb0966f2e97/src/Language/Haskell/GhciWrapper.hs#L63-L71

But things are actually even more complicated than that (not elaborating on it now, but if you want to work on it then please ask and I'll give the full story). That's why we are going with #448 for now.

File: src/GhcUtil.hs [hunk 1]
```diff
@@ -58,7 +62,18 @@ handleDynamicFlags flags = do
...
 #endif
-  (dynflags, locSrcs, _) <- (setHaddockMode `fmap` getSessionDynFlags) >>= (`parseDynamicFlags'` flags)
+  (dynflags', locSrcs, _) <- (setHaddockMode `fmap` getSessionDynFlags) >>= (`parseDynamicFlags'` flags)
+
+#if __GLASGOW_HASKELL__ >= 910
+  -- never display error links when displaying exception. Otherwise we get
+  -- failures like this:
+  --    expected: "Foo.hs:6:1: error: [GHC-58481]\n    parse error..."
+  --    but got: "Foo.hs:6:1: error: [\ESC]8;;https://errors.haskell.org/messages/GHC-58481\ESC\\GHC-58481\ESC]8;;\ESC\\]\n    parse error..."
+  let dynflags = dynflags' { useErrorLinks = Never }
```
Comment (sol) [inline src/GhcUtil.hs:72]: @hsyl20 we need to differentiate between:

1. Extracting comments with the GHCAPI.
2. Evaluating expressions with GHCi.

For (1) we want the exact same error behavior as GHC, including colors and hyperlinks.

For (2) we don't want colors and hyperlinks.54 OK

**Review by sol — CHANGES_REQUESTED** [review]
Please include only the version bump in this PR. (see my other comments)

**Comment by hsyl20** [comment]
I've removed the other commit. I won't have much time to work on this soon either.

**Pull request merged**